challenge_3/http.rb

```diff
@@ -0,0 +1,38 @@
+require 'socket'
+
+server = TCPServer.new 1234
+
+# "A request is what a browser will send to your application when you visit a
+#   URL like http://localhost:1234/my/awesome/path."
+
+loop do
+  # Accept a client connection
+  client = server.accept
+  puts "Got a new client!"
+  
+  # Read the request line
+  # Do I need the chomp? Seems to work fine without it
```

Comment:
Ah, interesting! That's a great point about encapsulating the reading of the line from the socket neatly in a single expression - if we were to move away from using `split` which automatically handles discarding the `CLRF` for us, something might break and it might not be clear why!

Reply:
Exactly. So it’s perhaps a kind of modularity / compositionality concern: it’s nice to be able to reason locally about each line of code and have confidence that it’s leaving everything in a consistent and unsurprising state, cos then you don’t have to remember any caveats when you come and make changes later.